src/jit/morph.cpp

```diff
@@ -12739,20 +12713,13 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
         {
             /* "x ^ -1" is "~x" */
             
-            if ((op2->gtOper == GT_CNS_INT) && (op2->gtIntConCommon.IconValue() == -1))
+            if (op2->IsIntegralConst(-1))
             {
                 tree->ChangeOper(GT_NOT);
                 tree->gtOp2 = NULL;
                 DEBUG_DESTROY_NODE(op2);
             }
-            else if ((op2->gtOper == GT_CNS_LNG) && (op2->gtIntConCommon.LngValue() == -1))
-            {
-                tree->ChangeOper(GT_NOT);
-                tree->gtOp2 = NULL;
-                DEBUG_DESTROY_NODE(op2);
-            }
-            else if ((op2->gtOper == GT_CNS_INT) && (op2->gtIntConCommon.IconValue() == 1) &&
-                     op1->OperIsCompare())
+            else if (op2->IsIntegralConst(1) && op1->OperIsCompare())
```

Comment:
Observation: Earlier we were performing this optimization only on TYP_INT on 32-bit and this change will perform on TYP_LONG as well.  Good change to extend to TYP_LONG.